src/fem/patch_fe_values.hh

```diff
@@ -22,20 +22,148 @@
 #define PATCH_FE_VALUES_HH_
 
 
-//#include <string.h>                           // for memcpy
-//#include <algorithm>                          // for swap
-//#include <new>                                // for operator new[]
-//#include <string>                             // for operator<<
-//#include <vector>                             // for vector
-//#include "fem/element_values.hh"              // for ElementValues
-//#include "fem/fe_values.hh"                   // for FEValuesBase
-//#include "fem/fe_values_views.hh"             // for FEValuesViews
-//#include "mesh/ref_element.hh"                // for RefElement
-//#include "mesh/accessors.hh"
-//#include "fem/update_flags.hh"                // for UpdateFlags
-//#include "quadrature/quadrature_lib.hh"
-
-
-
-//
+#include <string.h>                           // for memcpy
+#include <algorithm>                          // for swap
+#include <new>                                // for operator new[]
+#include <string>                             // for operator<<
+#include <vector>                             // for vector
+#include "fem/element_values.hh"              // for ElementValues
+#include "fem/fe_values.hh"                   // for FEValuesBase
+#include "fem/fe_values_views.hh"             // for FEValuesViews
+#include "mesh/ref_element.hh"                // for RefElement
+#include "mesh/accessors.hh"
+#include "fem/update_flags.hh"                // for UpdateFlags
+#include "quadrature/quadrature_lib.hh"
+#include "fields/eval_subset.hh"
+
+template<unsigned int spacedim> class PatchFEValues;
+
+
+
+using Scalar = double;
+using Vector = arma::vec3;
+using Tensor = arma::mat33;
+
+template <class ValueType>
+class ElQ {
+public:
+    /// Constructor
+    ElQ()
+    : fe_values_(nullptr), begin_(0) {}
+
+    /// Constructor
+    ElQ(PatchFEValues<3> *fe_values, unsigned int begin)
+    : fe_values_(fe_values), begin_(begin) {}
+
+    ValueType operator()(const BulkPoint &point)
```

Comment:
SidePoint je BulkPoint s dodatečnou informací o stěně, na které se nachází. Myslím, že bude stačit operátor pouze pro BulkPoint, protože pro výběr hodnoty z tabulky nebude informace o stěně potřeba.